### DISCUSSION TITLE
Option to force ZWave controller to only accept / allow S2 pairings?

### DISCUSSION THREAD

**Discussion by kinchahoy** [zwave-js/zwave-js#7204] [Feature Requests]
### Description

I mistyped a PIN and I think my Zooz ZEN20 got paired without authentication. I'd really like all my ZWave devices to be encrypted. Is there a way in the UI to force that (or to atleast make it hard to pair without S2 auth)?

### Do you have a solution in mind?

A setting to force S2 auth only pairing, and a special UI / button / option to ask the user if they want to allow an unsecured pairing

**Comment by AlCalzone** [comment]
The way the secure inclusion works, this isn't entirely possible. There should be a warning at the end of the process to notify you that the device was included insecurely though.
Security keys are exchanged after a node is part of the network, so if that fails, you'll have to remove the device and try again. Or use SmartStart, where this can be done automatically.
